### PR TITLE
feat: 0.4 | JS Fundamentals - Functions

### PR DESCRIPTION
**Pull request by beleshivprasad**

### CODE REVIEW

File: DAY4/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/*
+  Q.1 What happens when you add a for loop/while loop/switch case block inside a function and use return instead of break? Do statements after the loop run? What is the return value? Can we pass a return value from within a loop? Can you return from inside an if block? What impact does that have?
```
Comment (amar-josh) [inline DAY4/index.js:2]: while loop implementation is missing?

File: DAY4/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+test(callbackFunc()); // output: Calling the callback function and TypeError, callback is not function
+// callback hold values as 5 not function
+/*
+  as we are calling the function callbackFunc at the time of passing as argument to test, 
+  which will return value as 5 and  function test is trying to execute 5() and 5 is not a function so it is throwing error
+*/
+
+test(() => callbackFunc()); // output: Calling the callback function
+/*
+  we are passing callbackFunc inside the ans arrow function which is gettng executed and inside it we are executing callbackFunc and return value reutrned by callbackFunc
```
Comment (amar-josh) [inline DAY4/index.js:88]: reutrned => returned

**Review by amar-josh — APPROVED** [review]

File: DAY4/index.js (hunk 1 above)
Comment (beleshivprasad) [reply DAY4/index.js:2]: added implementation for while loop

File: DAY4/index.js (hunk 2 above)
Comment (beleshivprasad) [reply DAY4/index.js]: fixed

**Pull request merged**